seatunnel-core/seatunnel-core-base/src/main/java/org/apache/seatunnel/config/ConfigBuilder.java

```diff
@@ -157,6 +158,7 @@ private boolean checkIsContainHive() {
                 String serviceClassName = serviceClass.getName();
                 String clsNameToLower = serviceClassName.toLowerCase();
                 if (clsNameToLower.equals(canonicalName.toLowerCase())) {
+                    plugin.setConfig(config);
```

Comment:
@BenJFan Because config will be used by plugin

Reply:
> @BenJFan Because config will be used by plugin

Any problem with `plugin.setConfig` outside of `createPluginInstanceIgnoreCase` ?

Reply:
It would be better to pass object reference instead of value, which is more OOP style